src/components/generic/TimerComps.tsx

```diff
@@ -0,0 +1,289 @@
+                max={84400000}
+                defaultValue={3000}
+                onChange={event => {
+                    setDuration(Number.parseInt(event.target.value));
+                }}
+            />
+            <TimerInputComponent
+                label={labelRounds}
+                id="roundsInput"
+                min={1}
+                max={100}
+                defaultValue={3}
+                onChange={event => {
+                    setRounds(Number.parseInt(event.target.value));
+                }}
+            />
+        </TimerInputsContainer>
+    );
+};
+
+/*
+ * Props for the TabataTimerInput component.
+ */
+type TabataTimerInputProps = {
+    setDuration: (value: number) => void;
+    setRestDuration: (value: number) => void;
+    setRounds: (value: number) => void;
+    labelDuration?: string;
+    labelRestDuration?: string;
+    labelRounds?: string;
+};
+
+/*
+ * Component for the input of the time, rest time and rounds of the Tabata timer.
+ */
+export const TabataTimerInput = ({
+    setDuration,
+    setRounds,
+    setRestDuration,
+    labelDuration = 'Active Duration',
+    labelRestDuration = 'Rest Duration',
+    labelRounds = 'Rounds',
+}: TabataTimerInputProps) => {
+    return (
+        <TimerInputsContainer>
+            <TimerInputComponent
+                label={labelDuration}
+                id="timeInput"
+                min={1000}
+                max={84400000}
+                defaultValue={3000}
+                onChange={event => {
+                    setDuration(Number.parseInt(event.target.value));
+                }}
+            />
+            <TimerInputComponent
+                label={labelRestDuration}
+                id="restInput"
+                min={1000}
+                max={84400000}
+                defaultValue={2000}
+                onChange={event => {
+                    setRestDuration(Number.parseInt(event.target.value));
+                }}
+            />
+            <TimerInputComponent
+                label={labelRounds}
+                id="roundsInput"
+                min={1}
+                max={100}
+                defaultValue={3}
+                onChange={event => {
+                    setRounds(Number.parseInt(event.target.value));
+                }}
+            />
+        </TimerInputsContainer>
+    );
+};
+
+/*
+ * Component for the removal of a timer.
+ */
+export const RemoveButton = ({ removeId }: { removeId: string }) => {
+    const timersContext = useContext(TimersContext);
+    return <RemoveButtonStyle onClick={() => timersContext.deleteTimer(removeId)}>❌</RemoveButtonStyle>;
+};
+
+/*
+ * Component to display all the timers of the workout.
+ */
+export const TimersDisplay = ({ timers }: { timers: Timer[] }) => {
+    return (
+        <>
+            {timers.map(timer => (
+                <div key={timer.id}>
+                    {timer.type === 'Countdown' ? (
+                        <Countdown id={timer.id} />
```

Comment:
in general its not a good idea to use a prop called `id` as it might get confused with the HTML `id`. You can call it `timerId` here for example